addons/mail/static/src/core/web/follower_list.js

```diff
@@ -22,13 +24,33 @@ export class FollowerList extends Component {
         super.setup();
         this.action = useService("action");
         this.store = useState(useService("mail.store"));
+        this.followerListId = local_id++;
         useVisible("load-more", (isVisible) => {
             if (isVisible) {
-                this.props.thread.loadMoreFollowers();
+                this.followerListView.loadMoreFollowers();
+            }
+        });
+
+        onWillStart(async () => {
+            await this.store.FollowerListView.insert({
+                id: this.followerListId,
+                thread_id: this.props.thread.id,
+                thread_model: this.props.thread.model,
+            });
+            await this.followerListView.get_follower();
+        });
```

Comment:
Would move this to `onMounted()`.
`onWillStart` is blocking UI, which gives bad UX. So prefer to render earlier and re-render when data are available.

Also I heard some rumors `onWillStart()` may be removed. Personally from experience `onWillStart` is a bad smells, as sooner or later there are problems that are only solved by staying far away from `onWillStart`, so I would prefer to assume this hook doesn't exist.

Reply:
I did that only first, but it causes a flicker, as the component mounts first and then the followers get loaded

Reply:
You could load the follower list and open the dropdown only when this is loaded.
So yes, this means splitting JS component a bit different so the button itself is the component managing the load / loadMore of followers (and drop on close)

Reply:
I think loading of data should be done in `onWillStart` it self. Yes we can exclude the `await` to not block the UI and further rendering. With this the advantage we get is that while the component is being rendered and mounted we can utilize this time to fetch the data. Which helps in displaying updated data quicker with less flickers. Do let me know if I am missing something.